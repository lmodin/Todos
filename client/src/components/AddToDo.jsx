import React from 'react';

const AddToDo = (props) => {
  console.log(typeof props.addNewToDo)
  var newItem = ""

  const handleChange = (e) => {
    newItem = e.target.value;
    e.preventDefault();
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    props.addNewToDo(newItem);
  }

  return (
    <form className="addToDo" onSubmit={handleSubmit}>
      <input type="text" name="todo" default="Type a new To Do here" onChange={handleChange}></input>
      <input type="submit"></input>
    </form>
  )
}

export default AddToDo