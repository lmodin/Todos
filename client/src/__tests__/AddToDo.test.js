import React from 'react';
//import { cleanup, fireEvent, render} from '@testing-library/react';
import AddToDo from '../components/AddToDo.jsx';
import renderer from 'react-test-renderer';

//afterEach(cleanup);



//snapshot testing
it('renders correctly', () => {
  const form = renderer
    .create(<AddToDo ></AddToDo>)
    .toJSON();
  expect(form).toMatchSnapshot();
})