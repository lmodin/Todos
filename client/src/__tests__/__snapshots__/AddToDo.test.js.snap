// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="addToDo"
  onSubmit={[Function]}
>
  <input
    default="Type a new To Do here"
    name="todo"
    onChange={[Function]}
    type="text"
  />
  <input
    type="submit"
  />
</form>
`;
